src/md/bylaws.md

```diff
@@ -0,0 +1,381 @@
+
+###Voting
+
+Decisions regarding the project are made by votes on the primary project
+development mailing list (<a class="externalLink" href=
+"mailto:dev@apex.apache.org)">dev@apex.apache.org)</a>. Where necessary, PMC voting
+may take place on the private Apex PMC mailing list. Votes are clearly indicated by
+subject line starting with [VOTE]. Votes may contain multiple items for approval
+and these should be clearly separated. Voting is carried out by replying to the
+vote mail. Voting may take four flavours:
+
+<table border="0" class="table table-striped">
+<tbody>
+  <tr class="a">
+    <td>+1</td>
+
+    <td>&ldquo;Yes,&rdquo; &ldquo;Agree,&rdquo; or &ldquo;the action should be
+    performed.&rdquo; In general, this vote also indicates a willingness on the
+    behalf of the voter in &ldquo;making it happen&rdquo;</td>
+  </tr>
+
+  <tr class="b">
+    <td>+0</td>
+
+    <td>This vote indicates a willingness for the action under consideration to
+    go ahead. The voter, however will not be able to help.</td>
+  </tr>
+
+  <tr class="a">
+    <td>-0</td>
+
+    <td>This vote indicates that the voter does not, in general, agree with the
+    proposed action but is not concerned enough to prevent the action going
+    ahead.</td>
+  </tr>
+
+  <tr class="b">
+    <td>-1</td>
+
+    <td>This is a negative vote. On issues where consensus is required, this vote
+    counts as a veto. All vetoes must contain an explanation of why the veto is
+    appropriate. Vetoes with no explanation are void. It may also be appropriate
+    for a -1 vote to include an alternative course of action.</td>
+  </tr>
+</tbody>
+</table>
+
+All participants in the Apex project are encouraged to show their agreement with
+or against a particular action by voting. For technical decisions, only the votes
+of active committers are binding. Non binding votes are still useful for those with
+binding votes to understand the perception of an action in the wider Apex community.
+For PMC decisions, only the votes of PMC members are binding.
+
+Voting can also be applied to changes made to the Apex codebase. These typically
+take the form of a veto (-1) in reply to the commit message sent when the commit is
+made.
+
+###Approvals
+
+These are the types of approvals that can be sought. Different actions require
+different types of approvals:
+
+<table border="0" class="table table-striped">
+<tbody>
+  <tr class="a">
+    <td>Consensus</td>
+
+    <td>For this to pass, all voters with binding votes must vote and there can
+    be no binding vetoes (-1). Consensus votes are rarely required due to the
+    impracticality of getting all eligible voters to cast a vote.</td>
+  </tr>
+
+  <tr class="b">
+    <td>Lazy Consensus</td>
+
+    <td>Lazy consensus requires 3 binding +1 votes and no binding vetoes.</td>
+  </tr>
+
+  <tr class="a">
+    <td>Lazy Majority</td>
+
+    <td>A lazy majority vote requires 3 binding +1 votes and more binding +1
+    votes that -1 votes.</td>
```

Comment:
Does this mean binding +1 should be > binding+non-binding -1?

So, if there are 4 binding +1 and 5 non-binding -1 votes, does that mean the voting failed?